lib/util/error_code.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Util
+  # helps get information regarding current error codes.
+  class ErrorCode
+    def initialize(path)
+      @path = path
+      validate
+    end
+
+    def next_available
+      err_num = -1
+      File.foreach(@path) do |line|
+        match = line.match(/code: \"CONJ(?<num>[0-9]+)E\"/)
+        if match
+          err_num = [err_num, match[:num].to_i].max
+        end
+      end
```

Comment:
Two things here: 

If you were going to use this method, embrace the ruby way -- use the functional methods from the Enumerable class.  `select` all lines you want, `map` to pull the error numbers, then call `max`.  No procedural loops.

However, parsing the line text is brittle (eg, imagine someone refers to one of these error codes in comments, and has made a typo there).  Ruby allows you to introspect.  

`Errors::Conjur.constants` should give you all the Error class names, possibly others you don't care about.  Filter through them, seeing which respond to the `code` method.  Get all your codes that way.  Then apply the above logic to those strings.

Reply:
I chose the more brittle options due to the fact that I ran into issues loading the errors module. It would result in an error stating that it could not load `util/trackable_error_class` found in errors.rb.

Reply:
You should be able to just require it explicitly.

One of the most pernicious qualities of Rails is it's "convenient" auto-loading of classes.  This means you don't have to specify the things you're using with `require` statements at the top.  Which makes all your dependencies implicit, as well as screwing you when you use the files outside Rails.  I'm just guessing that's what's happening here (could be wrong).